### DISCUSSION TITLE
Vector database

### DISCUSSION THREAD

**Discussion by MrGold7** [Ironclad/rivet#131] [Q&A] [answered]
Hi, can you tell me how to work with vector database in this development environment. In brief, I need to connect Pinecone to a chat room

**Comment by abrenneke** [comment, marked as answer]
Hey check out these two graphs created by leop in the discord! They should give you examples of how to both do RAG retrieval using pinecone, and upset into pinecone!

[RAG_Time_FDA.rivet-graph](https://discord.com/channels/1149376303070466110/1149382754262339715/1151551211355516949)
[RAG_Time_Upsert_Loop.rivet-graph](https://discord.com/channels/1149376303070466110/1149382754262339715/1152303129262571653)
